src/loader.ts

```diff
@@ -0,0 +1,32 @@
+import assert from "node:assert";
+import type { LoadFnOutput, LoadHookContext } from "node:module";
+import { transformSync } from "./index.ts";
+
+type NextLoad = (
+	url: string,
+	context?: LoadHookContext,
+) => LoadFnOutput | Promise<LoadFnOutput>;
+
+export async function load(
+	url: string,
+	context: LoadHookContext,
+	nextLoad: NextLoad,
+) {
+	const { format } = context;
+	if (format.endsWith("-typescript")) {
+		// Use format 'module' so it returns the source as-is, without stripping the types.
+		// Format 'commonjs' would not return the source for historical reasons.
+		const { source } = await nextLoad(url, {
+			...context,
+			format: "module",
+		});
+		assert(source != null, "Source code cannot be null or undefined");
+		// biome-ignore lint/style/noNonNullAssertion: it's checked by the assertion above
+		const { code } = transformSync(source!.toString(), { mode: "strip-only" });
```

Comment:
We shouldn't need the non-null assertion if we have a runtime one
```suggestion
		assert(source != null, "Source code cannot be null or undefined");
		const { code } = transformSync(source.toString(), { mode: "strip-only" });
```
Alternatively, the following for sure works
```suggestion
		if(source == null) throw new Error("Source code cannot be null or undefined");
		const { code } = transformSync(source.toString(), { mode: "strip-only" });
```